docs/configuration-ini.md

```diff
@@ -101,159 +98,395 @@ beta2 = 0.999
 
-11. ```weight_initializer = <string>```
+Define the type and path of the validation data file.
+Start with "[valid_set]"
 
-     set weight initialization method
-       * zeros : Zero initialization
-       * lecun_normal : LeCun normal initialization
-       * lecun_uniform : LeCun uniform initialization
-       * xavier_normal : xavier normal initialization
-       * xavier_uniform : xavier uniform initialization
-       * he_normal : He normal initialization
-       * he_uniform : He uniform initialization
+1. ``` type = <string> ```
 
-12. ```filters = <unsigned int>```
+   Currently only file is supported.
 
-     set filters size for convolution layer
+2. ``` path = <string> ```
 
-13. ```stride = <unsigned int>,<unsigned int>```
+   Data path for validation.
 
-     set stride for convolution and pooling layer
+Below is a sample TrainSet section.
 
-14. ```padding = <unsigned int>,<unsigned int>```
+```ini
+# TrainSet Section
+[train_set]
+Type = file
+path = valDataset.dat
+```
 
-     set padding for convolution and pooling layer
+### Test Set Section
 
-15. ```pool_size = <unsigned int>,<unsigned int>```
+Define the type and path of the test data file.
+Start with "[valid_set]"
 
-     set pooling size for pooling layer
+1. ``` type = <string> ```
 
-16. ```pooling = <string>```
+   Currently only file is supported.
 
-     define type of pooling
-       * max : max pooling
-       * average : average pooling
-       * global_max : global max pooling
-       * global_average : global average pooling
+2. ``` path = <string> ```
 
-17. ```flatten = <bool>```
+   Data path for test.
 
-    flattens the output of this layer.
+Below is a sample TrainSet section.
 
-    Enabling this option is equivalent to attaching a flatten layer after the current layer.
+```ini
+# TrainSet Section
+[train_set]
+Type = file
+path = testDataset.dat
+```
+### Layer Section
 
-18. ```epsilon = <float>```
+Describe hyper-parameters for layer. Order of layers in the model follows the order of definition of layers here from top to bottom.
 
-    Epsilon parameter for batch normalization layer. Default is 0.001.
+Start with "[ ${layer name} ]". This layer name must be unique throughout network model.
 
-### Properties for layer
+1. ```type = <string>```
 
-Each layer requires different properties.
+   Type of Layer
+     * input : input layer
+     * fully_connected : fully connected layer
+     * batch_normalization : batch normalization layer
+     * conv2d : convolution 2D layer
+     * pooling2d : pooling 2D layer
+     * flatten : flatten layer
+     * activation : activation layer
+     * addition : addition layer
+     * concat : concat layer
+     * multiout : multiout layer
+     * embedding : embedding layer
+     * rnn : RNN layer
+     * lstm : LSTM layer
+     * split : split layer
+     * gru : GRU layer
+     * permute : permute layer
+     * dropout : dropout layer
+     * backbone_nnstreamer : backbone layer using nnstreamer
```

Comment:
NNTrainer also suppors backbone_tflite as well. HOw about adding it?

Reply:
okay